src/main/kotlin/com/rstudio/shinycannon/ProtectedApp.kt

```diff
@@ -24,10 +24,29 @@ import javax.xml.parsers.DocumentBuilderFactory
 import javax.xml.xpath.XPathConstants
 import javax.xml.xpath.XPathFactory
 
-data class HttpResponse(val statusCode: Int,
-                        val headers: Map<String, String>,
-                        val cookies: BasicCookieStore,
-                        val body: String)
+class HttpResponse(val statusCode: Int,
+                   val headers: Map<String, String>,
+                   val cookieStore: BasicCookieStore,
+                   val body: String) {
+    fun getCookie(name: String) = this.cookieStore.cookies.firstOrNull { it.name == name }
+    fun hasCookie(name: String) = this.getCookie(name) != null
+    fun getHeader(name: String) = this.headers.get(name)
+    fun hasHeader(name: String) = this.headers.containsKey(name)
```

Comment:
Perhaps `getHeader` and `hasHeader` should use `name.toLowerCase()` instead of `name`?